third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2585,7 +2684,38 @@ fn exp_(context: &mut Context, sp!(loc, pe_): P::Exp) -> E::Exp {
                 EE::BinopExp(exp(context, *pl), op, exp(context, *pr))
             }
         },
-        PE::Borrow(mut_, pr) => EE::Borrow(mut_, exp(context, *pr)),
+        PE::Borrow(mut_, pr) => {
+            if let PE::Index(ref e, ref i) = pr.value {
+                // handling `&_[_]` or `&mut _[_]`
+                if context.env.flags().v2() {
+                    if let PE::Name(sp!(_, ref ptn), _) = e.value {
+                        let name = ptn.name();
+                        if is_struct_in_current_module(context, name) {
```

Comment:
Once we have global structs, this won't work any longer. Is the problem constants that you can't make this decision just based on the name? In any case we can stay with this, we probably refactor once we remove expansion. (Which is the endgame to cut off from v1 compiler completely.)